mix/mix_protocol.nim

```diff
@@ -315,9 +329,10 @@ proc new*(
   mixProto.init()
   return ok(mixProto)
 
+# Do we need to raise error here?
 method init*(mixProtocol: MixProtocol) {.gcsafe, raises: [].} =
   proc handle(conn: Connection, proto: string) {.async: (raises: [CancelledError]).} =
     await mixProtocol.handleMixNodeConnection(conn)
 
-  mixProtocol.codecs = @[MixProtocolID]
-  mixProtocol.handler = handle
+  mixProtocol.lp_proto.codecs = @[MixProtocolID]
+  mixProtocol.lp_proto.handler = handle
```

Comment:
This is an example of composition over inheritence. It's much easier to see here that the codecs and handler are defined by the base protocol struct. no room for confusion between mix/base